libkirk/main.py

```diff
@@ -405,131 +405,124 @@ def run(cmd_args: list = None) -> None:
 
-    # generic arguments
-    parser.add_argument(
+    generic_opts = parser.add_argument_group('General options')
+    generic_opts.add_argument(
         "--version",
         "-V",
         action="version",
         version=f"%(prog)s, {__version__}")
-
-    # user interface arguments
-    parser.add_argument(
+    generic_opts.add_argument(
         "--verbose",
         "-v",
         action="store_true",
         help="Verbose mode")
-    parser.add_argument(
+    generic_opts.add_argument(
         "--no-colors",
         "-n",
         action="store_true",
         help="If defined, no colors are shown")
-
-    # generic directories arguments
-    parser.add_argument(
+    generic_opts.add_argument(
         "--tmp-dir",
         "-d",
         type=str,
         default="/tmp",
         help="Temporary directory")
-    parser.add_argument(
+    generic_opts.add_argument(
         "--restore",
-        "-R",
+        "-r",
         type=str,
         help="Restore a specific session")
+    generic_opts.add_argument(
+        "--json-report",
+        "-o",
+        type=str,
+        help="JSON output report")
 
-    # tests setup arguments
-    parser.add_argument(
+    conf_opts = parser.add_argument_group('Configuration options')
+    conf_opts.add_argument(
+        "--sut",
+        "-u",
+        default="host",
+        type=_sut_config,
+        help="System Under Test parameters. For help please use '--sut help'")
+    conf_opts.add_argument(
+        "--framework",
+        "-U",
+        default="ltp",
+        type=_framework_config,
+        help="Framework parameters. For help please use '--framework help'")
+    conf_opts.add_argument(
         "--env",
         "-e",
         type=_env_config,
         help="List of key=value environment values separated by ':'")
-    parser.add_argument(
+    conf_opts.add_argument(
         "--skip-tests",
-        "-k",
+        "-s",
         type=str,
         help="Skip specific tests")
-    parser.add_argument(
+    conf_opts.add_argument(
         "--skip-file",
-        "-K",
+        "-S",
         type=str,
         help="Skip specific tests using a skip file (newline separated item)")
-    parser.add_argument(
+
+    exec_opts = parser.add_argument_group('Execution options')
+    exec_opts.add_argument(
+        "--run-suite",
+        "-f",
+        nargs="*",
+        help="List of suites to run")
+    exec_opts.add_argument(
+        "--run-pattern",
+        "-p",
+        help="Run all tests matching the regex pattern")
+    exec_opts.add_argument(
+        "--run-command",
+        "-c",
+        help="Command to run")
+    exec_opts.add_argument(
         "--suite-timeout",
         "-T",
         type=_time_config,
         default="1h",
         help="Timeout before stopping the suite (default: 1h)")
-    parser.add_argument(
+    exec_opts.add_argument(
         "--exec-timeout",
-        "-t",
+        "-x",
```

Comment:
Why `-x`? IMHO not related to runltp or LTP library.

Reply:
`--runtime` is the option that was `-t` in `runltp` . I can use `-t` but then we need to change the other one